### PR TITLE
Fixes cardinality auth filter unit tests (#860)

### PR DESCRIPTION
**Pull request by unt01d**
This fixes the unit tests for the Cardinality/CardinalitySingle auth process filters.

The intention is to assert that certain requests result in a redirect.  In order to do this, we need to mock the HTTP helper class, and in order to do that we need to create a non-static HTTP wrapper which we can inject into the cardinality filter instances.

### CODE REVIEW

File: tools/phpunit/phpunit.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
             <exclude>
                 <directory>./../../vendor/</directory>
                 <directory>./../../tests/</directory>
+                <file>./../../lib/SimpleSAML/Utils/HttpAdapter.php</file>
```
Comment (unt01d) [inline tools/phpunit/phpunit.xml]: For some reason, the code coverage process was trying to include this file multiple times, which was killing it.  So the simplest thing to do was to ignore it (it doesn't need to be included in any case...).

**Comment by thijskh** [comment]
Testsuite now fails (again…) on psalm. I've committed a fix on master, maybe if you rebase on current master the tests will be green...

**Comment by thijskh** [comment]
@ghalse

**Comment by unt01d** [comment]
Done, but the 5.4 build seems to be failing now @thijskh...  :confused:

**Comment by thijskh** [comment]
That is an occasional transient problem in one test, fixable by restarting that job (which I did). Strange that coverage does not seem to report correctly anymore. But otherwise fine :+1:

Would like to see a comment from @ghalse since it mainly concerns his work.

**Comment by ghalse** [comment]
Sorry I've taken so long to respond - from what I've read recently, this certainly looks like the right sort of way to do things. And perhaps provides people with a useful hint on how to do unit tests for redirects later on :)

**Pull request merged**